Library/Homebrew/extend/os/linux/requirements/java_requirement.rb

```diff
@@ -0,0 +1,24 @@
+require "language/java"
+
+class JavaRequirement < Requirement
+  fatal true
+  default_formula "jdk"
+  download "http://www.oracle.com/technetwork/java/javase/downloads/index.html"
+
+  satisfy build_env: false do
+    jdk = Formula["jdk"]
+    @java_home = jdk.opt_prefix if jdk && jdk.installed?
+    next quiet_system "java", "-version"
```

Comment:
Looks like a regex to match `java -version`'s output should be pretty straight forward? Can help with that if requested.